core/src/smartcontracts/isi/account.rs

```diff
@@ -29,13 +29,19 @@ pub mod isi {
             wsv: &WorldStateView,
         ) -> Result<(), Self::Error> {
             let asset_id = self.object.id();
+            let account_id = asset_id.account_id.clone();
```

Comment:
I can't leave a comment for a proper line, but don't you think @appetrosyan meant `Register<AssetDefinition>` in #2097 rather than `Register<Asset>`? Btw I don't like that we even have `Register<Asset>`

Reply:
Glossary:

| Thing | What is it? | US dollar analogy |
|--------|--------|--------|
`Asset` | Particular asset value for particular account | US dollars in my bank deposit
`AssetDefinition` |  Declaration that an asset exists with such and such value type | US dollar as a currency
`Register<AssetDefinition>` | Create a new asset definition | Declare that there now exists a currency called US dollars (Coinage Act of 1792)
`Register<Asset>` | Initialize an asset value for particular account | Open a new deposit for US dollar in my bank account (maybe with some non-zero initial balance)
`Mint<Asset>` | Add some quantity to a particular `Asset` | US Treasury mints some dollars and puts them into my bank deposit (which sure would be nice) 

Before #2227 we didn't have such a thing as `Register<Asset>`. It was a representable instruction, but it wasn't implemented and errored out if one tried to execute it, which is why  #2227 was created in the first place.

After #2227 it does what's described in the table above: creates an `Asset` with some initial value. If `Asset` with the same `AssetDefinition` id is present for this account, it errors out. Crucially,  __it's not the only way to initialize an `Asset`__. If you make a transfer or mint to an account for which the asset isn't initialized, it will automatically be created.

Reply:
Thanks a lot for such a detailed explanation! It definitely wasn't clear to me which one would we want to modify in this PR despite going through the docs and code, so I went for a wrongful guess. Thus I agree that such a rename would be beneficial, and could incorporate it into this PR as it doesn't seem to warrant a separate issue, as long as @appetrosyan agrees.

Reply:
To me it seems like renaming `Asset` to `AssetBalance` and `AssetDefinition` to `Asset` will be pretty painful.  And not only for us, but for SDK devs too.

And I still don't understand why we can't just use `Mint<Asset>` and remove `Register<Asset>`

Reply:
To be clear, I'm not saying we should rename stuff, just saying how one can think about them to make it less confusing. I agree actually renaming this would be quite painful.

The problem with removing `Register<Asset>` is that it's easy to remove the implementation, but hard to make the instruction actually unrepresentable.

Reply:
> To me it seems like renaming Asset to AssetBalance and AssetDefinition to Asset will be pretty painful. And not only for us, but for SDK devs too.

So let's lump it in with the account restructuring. 

> The problem with removing Register<Asset> is that it's easy to remove the implementation, but hard to make the instruction actually unrepresentable.

We need a matrix of representable and implemented instructions.

Reply:
Tracking it with #3331.